spec/workers/cancel_previous_cycle_unsubmitted_applications_worker_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+RSpec.describe CancelPreviousCycleUnsubmittedApplicationsWorker do
+  include CycleTimetableHelper
+
+  describe '#perform' do
+    it 'cancels any unsubmitted applications from the last cycle' do
+      unsubmitted_application_from_last_year = create(
+        :application_form,
+        submitted_at: nil,
+        recruitment_cycle_year: RecruitmentCycle.previous_year,
+      )
+      create(
+        :application_choice,
+        status: :unsubmitted,
+        application_form: unsubmitted_application_from_last_year,
+        course_option: create(:course_option, course: create(:course, recruitment_cycle_year: RecruitmentCycle.previous_year)),
+      )
+
+      hidden_application_from_this_year = create(
+        :application_form,
+        submitted_at: nil,
+        candidate: create(:candidate, hide_in_reporting: true),
+        recruitment_cycle_year: RecruitmentCycle.current_year,
+      )
+      create(
+        :application_choice,
+        status: :unsubmitted,
+        application_form: hidden_application_from_this_year,
+        course_option: create(:course_option, course: create(:course, recruitment_cycle_year: RecruitmentCycle.current_year)),
+      )
+
+      unsubmitted_application_from_this_year = create(
+        :application_form,
+        submitted_at: nil,
+        recruitment_cycle_year: RecruitmentCycle.current_year,
+      )
+      create(
+        :application_choice,
+        status: :unsubmitted,
+        application_form: unsubmitted_application_from_this_year,
+        course_option: create(:course_option, course: create(:course, recruitment_cycle_year: RecruitmentCycle.current_year)),
+      )
+
+      rejected_application_from_this_year = create(
+        :application_form,
+        recruitment_cycle_year: RecruitmentCycle.current_year,
+      )
+      create(
+        :application_choice,
+        status: :rejected,
+        application_form: rejected_application_from_this_year,
+        course_option: create(:course_option, course: create(:course, recruitment_cycle_year: RecruitmentCycle.current_year)),
+      )
+
+      unsubmitted_cancelled_application_from_this_year = create(
+        :application_form,
+        submitted_at: nil,
+        recruitment_cycle_year: RecruitmentCycle.current_year,
+      )
+      create(
+        :application_choice,
+        status: :application_not_sent,
+        application_form: unsubmitted_cancelled_application_from_this_year,
+        course_option: create(:course_option, course: create(:course, recruitment_cycle_year: RecruitmentCycle.current_year)),
```

Comment:
I'm finding this test quite hard to read, you could use some let methods, e.g this course option is used four times, so we could have:

```suggestion
let(:course_option) { create(:course_option, course: create(:course, recruitment_cycle_year: RecruitmentCycle.current_year) } 
```

And/or perhaps some context blocks as well, like separating the different types of application form rejected, cancelled etc into its own context.

Reply:
I'm just working on the specs for the new worker – this one was copied over and renamed. Open to discussion on this but i'm not sure it's worth doing too much refactoring on this worker and corresponding specs, as there's a good chance we will remove this worker once we run it (which will happen once the PR is merged) – as going forward this will be a job in clock that will just run automatically for us at EoC. 

Reason this one was created was just because the manual task was missed (which automating it should prevent)!